docs/developer/agents.rst

```diff
@@ -430,3 +430,76 @@ containerized Agents together with the command
 
 Depending on your host's permissions, this command may need to be run with 
 ``sudo``.
+
+.. _timeout_lock
+
+TimeoutLock
+------------
+
+Overview
+^^^^^^^^^
+
+In OCS, operations are mainly asynchronous, however most agents have
+restrictions on which operations are allowed to be run simultaneously.
+For instance, when an Agent interfaces with a hardware device over serial
+communication, two commands cannot be issued simultaneously.
+In these cases, it may be required for agents to create a `lock` which restricts
+what operations can run concurrently.
+Requiring this lock to be acquired before any device communication is one
+way to ensure that serial messages are not crossed, or avoid any other
+dangerous asynchronous behavior.
+
+The ocs :ref:`TimeoutLock <ocs_twisted_api>` is a lock that implements a few useful additions to the
+standard ``threading.lock``. Mainly:
```

Comment:
Bullets in .rst aren't indented, this thows an "Unexpected indentation" warning when first compiled. Reformat with a blank line after "Mainly:" and no single space indication on the block.

https://docutils.sourceforge.io/docs/user/rst/quickref.html#bullet-lists